lib/task/tools/installTask.class.php

```diff
@@ -0,0 +1,536 @@
+        echo "Site description    {$siteOptions['siteDescription']}\n";
+        echo "Site base URL       {$siteOptions['siteBaseUrl']}\n";
+        echo "Admin email         {$adminOptions['email']}\n";
+        echo "Admin username      {$adminOptions['username']}\n";
+        echo "Admin password      {$adminOptions['password']}\n";
+
+        if (
+            !$options['no-confirmation']
+            && !$this->askConfirmation(
+                ['Confirm configuration and continue? (y/N)'],
+                'QUESTION_LARGE',
+                false
+            )
+        ) {
+            $this->logSection('install', 'Installation aborted');
+
+            exit(1);
+        }
+
+        $this->logSection('install', 'Setting configuration');
+
+        try {
+            arInstall::createDirectories();
+            arInstall::checkWritablePaths();
+            arInstall::createDatabasesYml();
+            arInstall::createPropelIni();
+            arInstall::createSettingsYml();
+            arInstall::createSfSymlink();
+            arInstall::configureDatabase($databaseOptions);
+            arInstall::configureSearch($searchOptions);
+        } catch (Exception $e) {
+            $this->logBlock(
+                [
+                    '',
+                    $e->getMessage(),
+                    '',
+                ],
+                'ERROR'
+            );
+
+            exit(2);
+        }
+
+        // Reload and test DB and ES configuration
+        $cacheClear = new sfCacheClearTask(
+            $this->dispatcher,
+            $this->formatter
+        );
+        $cacheClear->run();
+        Propel::configure($rootDir.'/config/config.php');
+        Propel::setDefaultDB('propel');
+        $this->configuration = ProjectConfiguration::getApplicationConfiguration(
+            'qubit',
+            'cli',
+            false
+        );
+        $this->context = sfContext::createInstance($this->configuration);
+        $this->context->databaseManager->loadConfiguration();
+        arElasticSearchPluginConfiguration::reloadConfig(
+            $this->context->getConfiguration()
+        );
+
+        try {
+            $this->context->getDatabaseConnection('propel');
+        } catch (Exception $e) {
+            $this->logBlock(
+                [
+                    '',
+                    'Database connection failure:',
+                    '',
+                    $e->getMessage(),
+                    '',
+                ],
+                'ERROR'
+            );
+
+            exit(2);
+        }
+
+        try {
+            arInstall::checkSearchConnection($searchOptions);
+        } catch (Exception $e) {
+            $this->logBlock(
+                [
+                    '',
+                    'Elasticsearch connection failure:',
+                    '',
+                    $e->getMessage(),
+                    '',
+                ],
+                'ERROR'
+            );
+
+            exit(2);
```

Comment:
Is there a meaningful difference between `exit(1)` used below and `exit(2)` used above?

Reply:
There are different exit/error codes, not something that we'd use but the general idea would be ...

- 0 = success
- 1 = aborted
- 2 = error

Reply:
Opinion: These should be constants if there is meaning behind them - e.g. `exit(self::ERROR)`

Reply:
I like that, thanks Steve.

Reply:
I'm thinking this doesn't add a lot of value considering that the error code could be useful in automated process and the difference between 1 and 2 would require user interaction to abort the task. So I think just `exit(1);` in all cases should be enough.

Reply:
Addressed in 0252ed5.